analyzers/src/SonarAnalyzer.CSharp/Rules/MethodsShouldNotHaveTooManyLines.cs

```diff
@@ -57,7 +58,8 @@ protected override void Initialize(ParameterLoadingAnalysisContext context)
         {
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
-                    if (c.ContainingSymbol.IsTopLevelMain()
+                    if ((c.Node.IsInTopLevelMain()
+                         && !c.Node.AncestorsAndSelf().OfType<MethodDeclarationSyntax>().Any())
```

Comment:
What is the use case for that? You registered for LocalFunctionStatement so c.Node must be a MethodDeclarationBaseSyntax. What am I missing here?